test/unit/math/mix/mat/prob/std_binormal_lcdf_test.cpp

```diff
@@ -0,0 +1,1245 @@
+  int j = 0;
+  for (int i = cntr; i < inp_vec.size(); ++i)
+    rho(j++) = (T(inp_vec(i)));
+}
+
+// Row vector, nada
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       Eigen::Matrix<T, 1, Eigen::Dynamic>& y) {
+  y(0) = inp_vec(0);
+  y(1) = inp_vec(1);
+}
+
+// vector<Vector>, nada
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, Eigen::Dynamic, 1>>& y) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+}
+
+// vector<Row Vector>, nada
+template <typename T>
+void to_function_input(int N_y,
+                       const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec,
+                       vector<Eigen::Matrix<T, 1, Eigen::Dynamic>>& y) {
+  int cntr = 0;
+  for (int i = 0; i < N_y; ++i) {
+    Eigen::Matrix<T, Eigen::Dynamic, 1> el(2);
+    el(0) = inp_vec(cntr);
+    ++cntr;
+    el(1) = inp_vec(cntr);
+    ++cntr;
+    y.push_back(el);
+  }
+}
+
+// Vector, real
+struct V_R_std_binorm_lcdf {
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    Matrix<T, Dynamic, 1> y(2);
+    y(0) = inp_vec(0);
+    y(1) = inp_vec(1);
+    return stan::math::std_binormal_lcdf(y, inp_vec(2));
+  }
+};
+
+// Row Vector, real
+struct RV_R_std_binorm_lcdf {
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    Matrix<T, 1, Dynamic> y(2);
+    y(0) = inp_vec(0);
+    y(1) = inp_vec(1);
+    return stan::math::std_binormal_lcdf(y, inp_vec(2));
+  }
+};
+
+// vector<Vector>, real
+struct vV_R_std_binorm_lcdf {
+  int N_y_;
+  explicit vV_R_std_binorm_lcdf(int N_y) : N_y_(N_y) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, Dynamic, 1>> y;
+    T rho;
+    to_function_input(N_y_, inp_vec, y, rho);
+    return stan::math::std_binormal_lcdf(y, rho);
+  }
+};
+
+// vector<Row Vector>, real
+struct vRV_R_std_binorm_lcdf {
+  int N_y_;
+  explicit vRV_R_std_binorm_lcdf(int N_y) : N_y_(N_y) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, 1, Dynamic>> y;
+    T rho;
+    to_function_input(N_y_, inp_vec, y, rho);
+    return stan::math::std_binormal_lcdf(y, rho);
+  }
+};
+
+// vector<Vector>, vector
+struct vV_v_std_binorm_lcdf {
+  int N_y_;
+  explicit vV_v_std_binorm_lcdf(int N_y) : N_y_(N_y) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, Dynamic, 1>> y;
```

Comment:
These four lines of code are repeated other than for type, so they should be pullable into a function.